documentation/specs/VerticalNav.md

```diff
@@ -0,0 +1,242 @@
+
+`VerticalNav.Item` is a polymorphic component. By default, it renders as an anchor element—`<a>`—but it can be rendered as a custom element using the `as` prop. e.g. `<VerticalNav.Item as={NextLink} href="/account/settings" />`.
+
+Selection and expansion is controlled by the consumer through the API provided by React Stately's `useListState` and `useTreeState` hooks—`selectedKey`, `expandedKeys`, and `onExpandedChange`.
+
+The component API is flexible by design. Constraints will be enforced at runtime to meet certain design requirements—e.g. limiting the levels of sub navigation.
+
+### API
+
+```ts
+type BaseVerticalNavProps = AriaLabelingProps & {
+  children: ReactNode;
+  renderBanner?: () => ReactNode;
+  renderLogo?: () => ReactNode;
+  supplementaryAction?: ReactNode;
+};
+
+type VerticalNavProps = BaseVerticalNavProps & {
+  selectedKey?: Key;
+};
+
+type ExpandableVerticalNavProps = BaseVerticalNavProps & {
+  selectedKey?: Key;
+  expandedKeys?: Key[];
+  onExpandedChange?: (keys: Key[]) => void;
+};
+
+type VerticalNavItemProps<T extends ElementType = "a"> = ComponentProps<T> & {
+  as?: T;
+  label: string;
+  icon?: IconSymbol;
+  children?: ReactNode;
+};
+
+type VerticalNavSubnavProps = {
+  selectedKey?: Key;
+};
+
+type VerticalNavSupplementaryAction<T extends ElementType = "button"> =
+  ComponentProps<T> & {
+    as?: T;
+    children: ReactNode;
+  };
+```
+
+### Example Usage
+
+_Simple vertical nav:_
+
+```tsx
+import { VerticalNav } from "@easypost/easy-ui/VerticalNav";
+
+function Sidebar() {
+  return (
+    <VerticalNav aria-label="Sidebar" selectedKey="1">
+      <VerticalNav.Item key="1" href="/item-1" icon={Symbol} label="Item 1" />
+      <VerticalNav.Item key="2" href="/item-2" icon={Symbol} label="Item 2" />
+      <VerticalNav.Item key="3" href="/item-3" icon={Symbol} label="Item 3" />
+      <VerticalNav.Item key="4" href="/item-4" icon={Symbol} label="Item 4" />
+    </VerticalNav>
+  );
+}
+```
+
+_Dense vertical nav:_
+
+```tsx
+import { VerticalNav } from "@easypost/easy-ui/VerticalNav";
+
+function Sidebar() {
+  return (
+    <VerticalNav aria-label="Sidebar" selectedKey="1">
+      <VerticalNav.Item key="1" href="/item-1" icon={Symbol} label="Item 1" />
+      <VerticalNav.Item key="2" href="/item-2" icon={Symbol} label="Item 2">
+        <VerticalNav.Subnav selectedKey="2a">
+          <VerticalNav.Item key="2a" href="/subitem-1" label="Subitem 1" />
+          <VerticalNav.Item key="2b" href="/subitem-2" label="Subitem 2" />
+        </VerticalNav.Subnav>
+      </VerticalNav.Item>
+      <VerticalNav.Item key="3" href="/item-3" icon={Symbol} label="Item 3" />
+      <VerticalNav.Item key="4" href="/item-4" icon={Symbol} label="Item 4" />
+    </VerticalNav>
+  );
+}
+```
+
+_Expandable vertical nav:_
```

Comment:
**question**: on mobile will the popover be implemented with a menu or listbox?

Reply:
good question. this may change after getting into the thick of implementation, but i think we can just keep it as the same implementation type (`nav`) on mobile since the behavior doesn't change from desktop to mobile. it would just be rendered as such in the container that changes to a "floating card" appearance